For your final project for this class, you have quite a bit of freedom, but the foundation must including generating a genome assembly, transcriptome assembly, or variant identification. This will culminate in a short presentation (10-15min) describing 

* your experimental design and question,
* your data source and analysis pipeline, 
* any pitfalls you encountered
* a summary of your results and conclusions

You must also turn in a minimum 3 page paper detailing these points, in addition to a section regarding "potential future directions" on how you would improve your experiment. 

If you would like, you can work with a partner on the analysis and interpretation of your data, and present jointly for the final project. You must **independently** write your own 3 page papers, though. Please ask me if you have any questions regarding potential plagiarism. 

# Potential Projects:

* Use SPAdes to assemble Illumina reads and annotate an _E. coli_ genome that produces Shiga toxin. This sample was isolated from faeces of diarrheic cattle. Compare your gene annotations to that of another E. coli strain. http://www.ncbi.nlm.nih.gov/sra/SRX1368955[accn]
* 

If you have your own data, you are encouraged to use that. We are here to help!