For your final project for this class, you have quite a bit of freedom, but the foundation must including generating a genome assembly, transcriptome assembly, or variant identification. This will culminate in a short presentation (10-15min) describing 

* your experimental design and question,
* your data source and analysis pipeline, 
* any pitfalls you encountered
* a summary of your results and conclusions



Some examples include:

* Assembling a _de novo_ transcriptome from an SRA experiment, annotating the transcripts against UniprotKB, and quantifying FPKMs for each gene. Extracting interesting genes for a biological question. 
* Assembling a _de novo_ bacterial genome from an SRA experiment, annotating the transcripts using Glimmer, and comparing your genome assembly and annotation to a closely related reference genome. Does adding PacBio data really improve a bacterial genome assembly, or is Illumina data enough?
* Using data from the 1000 Human Genomes Project, call SNP and indel polymorphisms for a single individual. What are the downsides to low coverage (3x) resequence data, but is it still usable?

If you have your own data, you are encouraged to use that. We are here to help!